rasterio/_base.pyx

```diff
@@ -1026,15 +1016,12 @@ cdef class DatasetBase:
         """
         m = Profile(**self.meta)
 
-        if self.is_tiled:
+        if self.block_shapes:
             m.update(
                 blockxsize=self.block_shapes[0][1],
                 blockysize=self.block_shapes[0][0],
-                tiled=True)
-        elif len(self.block_shapes) > 0:
-            m.update(blockysize=self.block_shapes[0][0], tiled=False)
-        else:
-            m.update(tiled=False)
+                )
+        m.update(tiled=self.block_shapes and self.block_shapes[0][1] != self.width)
```

Comment:
should this be `m.update(tiled=self.is_tiled)`?

Reply:
@groutr the subtext here is that `is_tiled` was a mistake. People (including us in this project) should use that one-liner instead.

Reply:
I see this PR changes `is_tiled` to:
```python
@property
def is_tiled(self):
    return self.block_shapes and self.block_shapes[0][1] != self.width
```
I'm not sure why we are duplicating the code.

Reply:
`is_tiled` was a mistake, but we can't remove it because other projects depend on it and it hasn't been deprecated. I'll add the deprecation in this PR.

Originally, `is_tiled` was mostly supposed to serve the `tiled` member of the `profile` property. And the only reason `tiled` is there is so that you can use a dataset's `profile` to create new datasets with similar structure. We don't do this for other creation options, but GTiff's `TILED` option is special in that it's what we almost always want today, but is not the default.

Reply:
Thank you for explaining in greater detail.